types/lookup_builder.go

```diff
@@ -8,26 +8,81 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"net/url"
 
 	"github.com/openfaas-incubator/connector-sdk/types"
-	"github.com/openfaas/faas/gateway/requests"
+	ptypes "github.com/openfaas/faas-provider/types"
+	"github.com/pkg/errors"
 )
 
 // FunctionLookupBuilder alias for types.FunctionLookupBuilder
 type FunctionLookupBuilder types.FunctionLookupBuilder
 
-// GetFunctions requests the OpenFaaS gteway to return a list of all functions
-func (lookup *FunctionLookupBuilder) GetFunctions() ([]requests.Function, error) {
-	req, _ := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/system/functions", lookup.GatewayURL), nil)
+//GetNamespaces get openfaas namespaces
+func (s *FunctionLookupBuilder) GetNamespaces() ([]string, error) {
+	var (
+		err        error
+		namespaces []string
+	)
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/system/namespaces", s.GatewayURL), nil)
+	if err != nil {
+		return namespaces, err
+	}
+
+	if s.Credentials != nil {
+		req.SetBasicAuth(s.Credentials.User, s.Credentials.Password)
+	}
+
+	res, err := s.Client.Do(req)
+	if err != nil {
+		return namespaces, err
+	}
+
+	if res.Body != nil {
+		defer res.Body.Close()
+	}
+
+	if res.StatusCode != http.StatusNotFound {
+		bytesOut, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return namespaces, err
+		}
+
+		if len(bytesOut) == 0 {
+			return namespaces, nil
+		}
+
+		err = json.Unmarshal(bytesOut, &namespaces)
+		if err != nil {
+			return namespaces, err
+		}
+	}
+
+	return namespaces, err
+}
+
+// GetFunctions requests the OpenFaaS gateway to return a list of all functions
+func (s *FunctionLookupBuilder) GetFunctions(namespace string) ([]ptypes.FunctionStatus, error) {
```

Comment:
Please use CLI SDK for this API calls, try to vendor github.com/openfaas/faas-cli/proxy and use API to functions for a namespace .